libs/k8s-client/src/ApiClient/PodsApiClient.php

```diff
@@ -31,22 +32,32 @@ public function getStatus(string $name, array $queries = []): Pod
         return $this->apiClient->request($this->baseApi, 'readStatus', Pod::class, $name, $queries);
     }
 
-    public function readLog(string $name, array $queries = []): string
+    public function readLog(string $name, array $queries = []): StreamInterface
     {
-        // do not use $this->request() wrapper, method does not return a class
-        $result = $this->baseApi->readLog($this->apiClient->getK8sNamespace(), $name, $queries);
+        $client = StreamClient::getInstance();
+        $response = $client->request(
+            'get',
+            sprintf('/api/v1/namespaces/%s/pods/%s/log', $this->apiClient->getK8sNamespace(), $name),
+            [
+                'query' => $queries,
+            ],
+        );
 
-        if ($result instanceof Status) {
-            throw new KubernetesResponseException(sprintf('K8S request has failed: %s', $result->message), $result);
-        }
+        if ($response->getStatusCode() >= 400) {
```

Comment:
Na callu jsme se bavili, ze pokud request neskonci 2xx, mel by se provest standardni processing response, ktery se to pokusi napasovat na `Status` class